src/ripple/app/tx/impl/Escrow.cpp

```diff
@@ -521,11 +516,27 @@ EscrowCancel::doApply()
     if (! slep)
         return tecNO_TARGET;
 
-    // Too soon?
-    if (! (*slep)[~sfCancelAfter] ||
-        ctx_.view().info().parentCloseTime.time_since_epoch().count() <=
+    if (ctx_.view ().rules().enabled(fix1523))
+    {
+        auto const now = ctx_.view().info().parentCloseTime;
+
+        // No cancel time specified: can't execute at all.
+        if (!(*slep)[~sfCancelAfter])
```

Comment:
This is tricky because it changes the semantics of existing escrows. Ideally, we'd like the new rules to apply to escrows created with the new rules. I don't know a good way to do this, and it's probably fine in this case.

Reply:
We definitely want it to apply to all escrows. The old semantics were broken, hence the motivation for the amendment.

Reply:
This is my bad, I'm actually misread the code to think it was a bigger change than it is (my fault, not the code's).